migration_pipeline.py

```diff
@@ -88,15 +102,67 @@ def build_import_pipeline(client_config, checkpoint_service, args) -> Pipeline:
     import_instance_profiles = pipeline.add_task(InstanceProfileImportTask(client_config))
     import_users = pipeline.add_task(UserImportTask(client_config), [import_instance_profiles])
     import_groups = pipeline.add_task(GroupImportTask(client_config), [import_users])
-    import_notebooks = pipeline.add_task(NotebookImportTask(client_config, checkpoint_service, args), [import_groups])
-    import_workspace_acls = pipeline.add_task(WorkspaceACLImportTask(client_config, checkpoint_service), [import_notebooks])
+    import_notebooks = pipeline.add_task(
+        NotebookImportTask(client_config, checkpoint_service, args), [import_groups])
+    import_workspace_acls = pipeline.add_task(
+        WorkspaceACLImportTask(client_config, checkpoint_service), [import_notebooks])
     import_secrets = pipeline.add_task(SecretImportTask(client_config), [import_groups])
     import_clusters = pipeline.add_task(ClustersImportTask(client_config, args), [import_secrets])
-    import_instance_pools = pipeline.add_task(InstancePoolsImportTask(client_config, args), [import_clusters])
+    import_instance_pools = pipeline.add_task(InstancePoolsImportTask(client_config, args),
+                                              [import_clusters])
     import_jobs = pipeline.add_task(JobsImportTask(client_config, args), [import_instance_pools])
-    import_metastore = pipeline.add_task(MetastoreImportTask(client_config, checkpoint_service, args), [import_groups])
-    import_metastore_table_acls = pipeline.add_task(MetastoreTableACLImportTask(client_config, args), [import_metastore])
+    import_metastore = pipeline.add_task(
+        MetastoreImportTask(client_config, checkpoint_service, args), [import_groups])
+    import_metastore_table_acls = pipeline.add_task(
+        MetastoreTableACLImportTask(client_config, args), [import_metastore])
+
+    return pipeline
+
+
+def build_validate_pipeline(client_config, checkpoint_service, args):
+    completed_pipeline_steps = checkpoint_service.get_checkpoint_key_set(
+        wmconstants.WM_VALIDATE, wmconstants.MIGRATION_PIPELINE_OBJECT_TYPE)
+
+    base_dir = client_config['base_dir']
+    source_dir = os.path.join(base_dir, args.source_session) + '/'
+    destination_dir = os.path.join(base_dir, args.destination_session) + '/'
+
+    pipeline = Pipeline(client_config['export_dir'], completed_pipeline_steps, args.dry_run)
 
+    def add_diff_task(name, config, file_path, parents=None):
+        source_file = os.path.join(source_dir, file_path)
+        destination_file = os.path.join(destination_dir, file_path)
+        return pipeline.add_task(DiffTask(name, source_file, destination_file, config), parents)
+
+    add_diff_task(
+        "validate-users",
+        DiffConfig(
+            primary_key='userName',
+            ignored_keys={'id', 'entitlements'},
```

Comment:
can you leave comments on why entitlements is also ignored?

Reply:
Reverted.